packages/prop-controllers/src/gap-x/gap-x.ts

```diff
@@ -1,16 +1,21 @@
 import { Types } from '../prop-controllers'
 import { GapData, responsiveGapDataSchema } from '../data'
-import { versionedPropDef } from '../versioned'
+import { versionedPropDef, typeArg } from '../versioned'
 
-export const GapX = versionedPropDef<{
-  preset?: unknown
-  label?: string
-  defaultValue?: GapData
-  min?: number
-  max?: number
-  step?: number
-  hidden?: boolean
-}>()(Types.GapX, responsiveGapDataSchema, {
-  version: 1,
-  dataKey: 'prop-controllers::gap-x::v1',
-})
+export const GapX = versionedPropDef(
+  Types.GapX,
+  responsiveGapDataSchema,
+  {
+    version: 1,
+    dataKey: 'prop-controllers::gap-x::v1',
+  },
+  typeArg<{
+    preset?: unknown
+    label?: string
+    defaultValue?: GapData
+    min?: number
+    max?: number
+    step?: number
+    hidden?: boolean
+  }>(),
```

Comment:
This is a very interesting API. Particularly because it's a type-only construct. I've had some bad experience in TypeScript when I've had to manually pass types on generics or have had to cast types. Particularly around type safety and discrepancies between run time and compile time. I wonder if there are any gotchas with this API and if we could enforce it at run time by taking an options Zod schema.

Reply:
We could certainly validate options at run-time if we want to. My understanding was that options are specified exclusively in the source code, though, and thus compile-time validation is sufficient.